packages/evolution-common/src/services/baseObjects/BaseInterview.ts

```diff
@@ -69,21 +70,73 @@ export class BaseInterview extends Surveyable implements IValidatable {
      * @param dirtyParams
-     * @returns BaseInterview
+     * @returns BaseInterview | Error[]
      */
-    static create(dirtyParams: { [key: string]: any }) {
-        const allParamsValid = true;
-        // validate types and required attributes
-        // TODO: implement dirtyParams validation:
-        const params: BaseInterviewAttributes = {
-            sample: dirtyParams.sample,
-            survey: dirtyParams.survey
-        };
-        if (allParamsValid === true) {
-            return new BaseInterview(params);
+    static create(dirtyParams: { [key: string]: any }): BaseInterview | Error[] {
+        const errors = BaseInterview.validateParams(dirtyParams);
+        return errors.length > 0 ? errors : new BaseInterview(dirtyParams as ExtendedInterviewAttributes);
+    }
+
+    /**
+     * Validates attribute types for BaseInterview.
+     * @param dirtyParams The parameters to validate.
+     * @returns An array of validation errors.
+     */
+    static validateParams(dirtyParams: { [key: string]: any }): Error[] {
+        const errors: Error[] = [];
+
+        // Validate params object:
+        if (!dirtyParams || typeof dirtyParams !== 'object') {
+            errors.push(new Error('BaseInterview validateParams: params is undefined or invalid'));
+            return errors; // Stop now, as the params are not valid.
+        }
+
+        // Validate UUID
+        const uuidErrors: Error[] = Uuidable.validateParams(dirtyParams);
+        if (uuidErrors.length > 0) {
+            errors.push(...uuidErrors);
+        }
+
+        // Validate startedDate (if provided)
+        if (dirtyParams.startedDate !== undefined && (!(dirtyParams.startedDate instanceof Date) || isNaN(dirtyParams.startedDate.getDate()))) {
+            errors.push(new Error('BaseInterview validateParams: invalid startedDate'));
+        }
+
+        // Validate startedTime (if provided)
+        if (dirtyParams.startedTime !== undefined && (typeof dirtyParams.startedTime !== 'number' || dirtyParams.startedTime < 0)) {
+            errors.push(new Error('BaseInterview validateParams: startedTime should be a non-negative number'));
+        }
+
+        // Validate completedDate (if provided)
+        if (dirtyParams.completedDate !== undefined && (!(dirtyParams.completedDate instanceof Date) || isNaN(dirtyParams.completedDate.getDate()))) {
+            errors.push(new Error('BaseInterview validateParams: invalid completedDate'));
+        }
+
+        // Validate completedTime (if provided)
+        if (dirtyParams.completedTime !== undefined && (typeof dirtyParams.completedTime !== 'number' || dirtyParams.completedTime < 0)) {
+            errors.push(new Error('BaseInterview validateParams: completedTime should be a non-negative number'));
+        }
+
+        // Validate baseHousehold (if provided)
+        if (dirtyParams.baseHousehold !== undefined && !(dirtyParams.baseHousehold instanceof BaseHousehold)) {
```

Comment:
Shouldn't the interview be responsible to construct its own household? Now the dirty params seems to come with the household, which comes with the persons and trips and segments, etc. That means you have to dig down the object hierarchy to create to deepest objects first. You might want to let the interview create the household, and the household create the persons from the dirty params. The factory function would create the contained objects first

Reply:
Weighting a sample over the reference population is the usual process involved when dealing with statistically based surveys. We need to make sure that the results are representative of the overall population associated with the survey. Each main object of survey data can be weighted, and each can include distinct weighting methods, so this is why we need to accept an array of weights, which includes the weight itself (a decimal number) and the weighting method. I am not sure we should include this in the documentation for now, since this is pretty usual stuff. All Origin-Destination and travel surveys are weighted.